kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/resolver/DependencyResolver.java

```diff
@@ -0,0 +1,104 @@
+        Map<String, GenericKubernetesResource> filters = context.getSecondaryResources(GenericKubernetesResource.class).stream().collect(toByNameMap());
+        var resolutionResult = virtualKafkaClusters.stream().map(cluster -> {
+            VirtualKafkaClusterSpec spec = cluster.getSpec();
+            Set<UnresolvedDependency> unresolvedDependencies = new HashSet<>();
+            spec.getIngressRefs().stream()
+                    .filter(ref -> !ingresses.containsKey(ref.getName()))
+                    .map(ref -> new UnresolvedDependency(KAFKA_PROXY_INGRESS, ref.getName())).forEach(unresolvedDependencies::add);
+            String clusterRef = spec.getTargetCluster().getClusterRef().getName();
+            if (!clusterRefs.containsKey(clusterRef)) {
+                unresolvedDependencies.add(new UnresolvedDependency(KAFKA_CLUSTER_REF, clusterRef));
+            }
+            List<Filters> filtersList = spec.getFilters();
+            if (filtersList != null) {
+                filtersList.stream()
+                        .filter(filterRef -> filters.values().stream().noneMatch(filterResource -> filterResourceMatchesRef(filterRef, filterResource)))
+                        .map(ref -> new UnresolvedDependency(FILTER, ref.getName())).forEach(unresolvedDependencies::add);
+            }
+            return new ResolutionResult.ClusterResolutionResult(cluster, unresolvedDependencies);
+        }).collect(Collectors.toMap(result -> ResourcesUtil.name(result.cluster()), r -> r));
+        ResolutionResult result = new ResolutionResult(filters, ingresses, clusterRefs, resolutionResult);
+        reportClustersThatDidNotFullyResolve(context, result);
+        return result;
+    }
+
+    private static boolean filterResourceMatchesRef(Filters filterRef, GenericKubernetesResource filterResource) {
+        String apiVersion = filterResource.getApiVersion();
+        var filterResourceGroup = apiVersion.substring(0, apiVersion.indexOf("/"));
+        return filterResourceGroup.equals(filterRef.getGroup())
+                && filterResource.getKind().equals(filterRef.getKind())
+                && name(filterResource).equals(filterRef.getName());
+    }
+
+    public static void reportClustersThatDidNotFullyResolve(Context<KafkaProxy> context, ResolutionResult resolutionResult) {
+        resolutionResult.clusterResult()
+                .filter(ResolutionResult.ClusterResolutionResult::isAnyDependencyUnresolved)
+                .forEach(clusterResolutionResult -> {
+                    clusterResolutionResult.firstUnresolvedDependency().ifPresent(unresolved -> {
+                        VirtualKafkaCluster cluster = clusterResolutionResult.cluster();
+                        switch (unresolved.type()) {
+                            case KAFKA_PROXY_INGRESS -> {
+                                SharedKafkaProxyContext.addClusterCondition(context, cluster, ingressNotFound(name(cluster), unresolved.name()));
```

Comment:
I'd refactor to so that we are calling a callback rather than invoking SharedKafkaProxyContext.addClusterCondition directly.

Reply:
done, have introduced an `UnresolvedDependencyReporter` callback